nixos/modules/services/web-apps/gancio.nix

```
@@ -0,0 +1,253 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.gancio;
+  json = pkgs.formats.json { };
+in
+{
+  options.services.gancio = {
+    enable = lib.mkEnableOption "Gancio, a shared agenda for local communities";
+
+    package = lib.mkPackageOption pkgs "gancio" { };
+
+    plugins = lib.mkOption {
+      type = with lib.types; listOf package;
+      example = lib.literalExpression "[ pkgs.gancioPlugins.telegram-bridge ]";
+      description = ''
+        Paths of gancio plugins to activate (linked under $WorkingDirectory/plugins/).
+      '';
+    };
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      description = "The user (and database name) used to run the gancio server";
+      default = "gancio";
+    };
+
+    hostName = lib.mkOption {
+      type = lib.types.str;
+      default = "localhost";
+      description = "The domain under which gancio will be reachable via nginx";
+    };
+
+    listenHost = lib.mkOption {
+      type = lib.types.str;
+      default = "localhost";
+      example = "::";
+      description = ''
+        The address (IPv4, IPv6 or DNS) for the gancio server to listen on.
+      '';
+    };
+
+    listenPort = lib.mkOption {
+      type = lib.types.port;
+      default = 13120;
+      description = ''
+        Port number of the gancio server to listen on.
+      '';
+    };
+
+    dbDialect = lib.mkOption {
+      type = lib.types.enum [
+        "sqlite"
+        "postgres"
+      ];
+      default = "sqlite";
+      description = ''
+        The database dialect to use
+      '';
+    };
+
+    logLevel = lib.mkOption {
+      description = "Gancio log level.";
+      type = lib.types.enum [
+        "debug"
+        "info"
+        "warning"
+        "error"
+      ];
+      default = "info";
+    };
+
+
+    settings = lib.mkOption rec {
+      type = json.type;
+      apply = lib.recursiveUpdate default;
+      default = {
+        baseurl = "http${
+          lib.optionalString config.services.nginx.virtualHosts."${cfg.hostName}".enableACME "s"
+        }://${cfg.hostName}";
+        server = {
+          host = cfg.listenHost;
+          port = cfg.listenPort;
+        };
+        log_level = cfg.logLevel;
+        log_path = "/var/log/${cfg.user}";
+        db =
+          {
+            dialect = cfg.dbDialect;
+          }
+          // lib.optionalAttrs (cfg.dbDialect == "sqlite") { storage = "/var/lib/${cfg.user}/db.sqlite"; }
```

```suggestion
          // lib.optionalAttrs (cfg.dbDialect == "sqlite") { storage = "/var/lib/gancio/db.sqlite"; }
```
thats wrong. /var/lib contains the service name, not the user it is running